tests/application_development/code_relocation/src/main.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <ztest.h>
+
+/* This function will allow execute from sram region.
+ * This is needed only for this sample because by default all soc will
+ * disable the execute from SRAM.
+ * An application that requires that the code be executed from SRAM will
+ *  have to configure the region appropriately in arm_mpu_regions.c.
+ */
+
+#if (defined(CONFIG_ARM_MPU) && !defined(CONFIG_CPU_HAS_NXP_MPU))
+#include <zephyr/arch/arm/aarch32/cortex_m/cmsis.h>
+void disable_mpu_rasr_xn(void)
+{
+	uint32_t index;
+	/* Kept the max index as 8(irrespective of soc) because the sram
+	 * would most likely be set at index 2.
+	 */
+	for (index = 0U; index < 8; index++) {
+		MPU->RNR = index;
+#if defined(CONFIG_ARMV8_M_BASELINE) || defined(CONFIG_ARMV8_M_MAINLINE)
+		if (MPU->RBAR & MPU_RBAR_XN_Msk) {
+			MPU->RBAR ^= MPU_RBAR_XN_Msk;
```

Comment:
May I ask why using the `^` here?

Reply:
means unlock all mpu regions from 0 address, and keeps other settings

Reply:
Sorry, I meant any special reasons that why not using `MPU->RBAR &= ~MPU_RBAR_XN_Msk;` since it is going to disable the XN bit?

Reply:
it is operated when the xn bit is set, as this is only one bit, pre my understanding xor is one atomic operation, but ~, & are two bit operations.